packages/core/src/lib/implementation/execute-plugin.spec.ts

```diff
@@ -129,6 +129,33 @@ describe('executePlugins', () => {
     ).rejects.toThrow('Audit metadata not found for slug mock-audit-slug');
   });
 
+  it('should log invalid plugin errors and throw', async () => {
+    const pluginConfig = {
+      ...validPluginCfg,
+      runner: vi.fn().mockRejectedValue('plugin 1 error'),
+    };
+    const pluginConfig2 = {
+      ...validPluginCfg2,
+      runner: vi.fn().mockResolvedValue([]),
+    };
+    const pluginConfig3 = {
+      ...validPluginCfg,
+      runner: vi.fn().mockRejectedValue('plugin 3 error'),
+    };
+    const plugins = [pluginConfig, pluginConfig2, pluginConfig3];
+    const errorSpy = vi.spyOn(console, 'error');
```

Comment:
The spy should be restored, so that `console.error` then reverts to its original implementation.

You could put `errorSpy.mockRestore()` at the end of the `it`-block, but then it will only be run when all assertions pass (otherwise it wouldn't be reached).

A better way is to use hooks, which the test runner will always execute before/after each/all tests:

```ts
describe('test suite', () => {
  let errorSpy: SpyInstance;
  
  beforeEach(() => {
    errorSpy = vi.spyOn(console, 'error');
  });
  
  afterEach(() => {
    errorSpy.mockRestore();
  });
  
  it('test 1', () => ...);

  it('test 2', () => ...);

  // ...
})
```

Reply:
Fixed, thanks